### PR TITLE
feat: add keyboardHidesTabBarAnimationConfig props

### LINKED ISSUE #149: keyboardHidesTabBar should instantly hide tab bar
### Current behaviour
Currently it animates down

### Expected behaviour
I would like it to instantly disappear to improve the experience.

### PR DESCRIPTION
**Pull request by adrianha**
### Motivation

Fixes #149 
Add `keyboardHidesTabBarAnimationConfig` on `TabBarOptions` to modify keyboard animation

### Test plan

Use this config as `TabBarOptions`:
```
tabBarOptions: {
  keyboardHidesTabBar: true,
  keyboardHidesTabBarAnimationConfig: {
    show: {
      animation: 'timing' | 'spring',
      config?: TimingKeyboardAnimationConfig | SpringKeyboardAnimationConfig
    },
    hide: {
      animation: 'timing' | 'spring',
      config?: TimingKeyboardAnimationConfig | SpringKeyboardAnimationConfig
    }
  }
}
```

### CODE REVIEW

**Comment by satya164** [comment]
I'd rather have something like `keyboardHidesTabBarAnimationConfig` which provides a custom animation e.g. `{ animation: timing, config: { duration: 0 } }` so user can also adjust the animation.

**Comment by adrianha** [comment]
Noted. So i think there will be 2 configs for animations show and hide. What do you think @satya164 ?

**Comment by satya164** [comment]
I think we can make it an object with open/close properties. We could have default value for both and shallow merge.

**Comment by adrianha** [comment]
Yeah, will update the PR

**Comment by adrianha** [comment]
just updated the config, but its only support `timing` animation for now. Any advice how to define types if i want to add `decay` or `spring` animation? since the config is different, eg: `decay` need `velocity` and doesnt have `duration` as a config. cc @satya164

**Comment by adrianha** [comment]
after giving a thought, i guess `timing` is the only possible animation type for keyboard show/hide animation. Could we use `decay` or `spring` animation to handle this? i think custom `duration` should be enough. maybe something like this:
```
keyboardHidesTabBarAnimationConfig: {
  show: {
    duration: 150
  },
  hide: {
    duration: 100
  }
}
```
What do you think @satya164 ?

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
   "homepage": "https://github.com/react-navigation/tabs#readme",
   "dependencies": {
     "hoist-non-react-statics": "^3.3.0",
+    "lodash": "^4.17.15",
```
Comment (satya164) [inline package.json]: I'd like to not use lodash. This should be enough:

```js
{ ...defaultConfig, ...userConfig }
```

**Comment by adrianha** [comment]
Just updated the PR
- Update type definitions; now support `timing` and `spring` animation
- Remove lodash

**Comment by adrianha** [comment]
Hi @satya164 , any suggestion or concern regarding the new updates?

**Comment by satya164** [comment]
Thanks for the updates!

**Pull request merged**